tf2_filter/include/tf2_filter/tf2_filter_config.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2018 Robert Bosch GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef TF2_FILTER__TF2_FILTER_CONFIG_HPP_
+#define TF2_FILTER__TF2_FILTER_CONFIG_HPP_
+
+#include <string>
+#include <vector>
+#include <tf2_filter/tf2_filter.hpp>
+#include <rclcpp/parameter.hpp>
+
+namespace tf2_filter
+{
+/** Create a filter map from the list in param.
+ * The value is expected to be a string array (see filtermap_from_frames).
+*/
+FilterMap filtermap_from_param(const rclcpp::Parameter & param);
+
+/**
+ * Create a filter map from the list of frames.
+ * Frame ids must start with the top-level parent frame and be in
+ * in the order of the TF tree, and there must be at least two frame ids.
+ */
+FilterMap filtermap_from_frames(const std::vector<std::string> & frame_ids);
```

Comment:
My preference would be to do the computation of the chain as an external process, e.g., as part of the launch process.

Moreover, my intention has been to offer a service to be able to change the chain at runtime.

I can explain the rationale in more detail in the design docs, but the gist is that 1) it drastically simplifies both the implementation and the runtime resource consumption of the filter when it doesn't have to keep a TF buffer around and 2) at start-up, there may be a period when the chain is not yet fully known. This would require either waiting until it is, or passing on everything for some short time, both of which seem to be not ideal.

By making this an external step, we can provide multiple options later.